api/client/accesslist/accesslist.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accesslist
+
+import (
+	"context"
+
+	"github.com/gravitational/trace/trail"
+
+	accesslistv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/accesslist/v1"
+	"github.com/gravitational/teleport/api/types/accesslist"
+	conv "github.com/gravitational/teleport/api/types/accesslist/convert/v1"
+)
+
+// Client is an access list client that conforms to the following lib/services interfaces:
+// * services.AccessLists
```

Comment:
I'd add a `var _ AccessLists = (*accesslist.Client)(nil)` in `lib/services` or something like that, to enforce the interface; either that, or move the interface somewhere in `api` so you can do that here.